modules/wechat_qrcode/src/zxing/qrcode/decoder/decoded_bit_stream_parser.cpp

```diff
@@ -190,16 +191,20 @@ void DecodedBitStreamParser::decodeByteSegment(Ref<BitSource> bits_, string& res
                                                CharacterSetECI* currentCharacterSetECI,
                                                ArrayRef<ArrayRef<char> >& byteSegments,
                                                ErrorHandler& err_handler) {
-    int nBytes = count;
     BitSource& bits(*bits_);
     // Don't crash trying to read more bits than we have available.
     int available = bits.available();
     // try to repair count data if count data is invalid
     if (count * 8 > available) {
-        count = (available + 7 / 8);
+        count = (available + 7) / 8;
     }
+    size_t nBytes = count;
```

Comment:
Maybe we should add something here...

```cpp
if (count <= 0) {
    // Is it necessary to assign a FormatErrorHandler to err_handler?
    err_handler = zxing::FormatErrorHandler("decodeByteSegment");
    return;
}
```

Reply:
`count=0` is also a legal ByteSegment, no need to throw error exception.